projects/python-contract-examples/smart_contracts/subsidize_app_call/contract.py

```diff
@@ -0,0 +1,34 @@
+from algopy import (
+    Application,
+    Bytes,
+    Global,
+    TemplateVar,
+    TransactionType,
+    Txn,
+    UInt64,
+    logicsig,
+)
+from algopy.op import GTxn
+
+
+@logicsig
+def subsidize_app_call() -> bool:
+    """
+    This Contract Account will subsidize the fees for any AppCall transaction directed to a known application.
+    """
+    return (
+        # is it safe to pay for the fees of the previous transaction?
+        Txn.type_enum == TransactionType.Payment
+        and Txn.receiver == Txn.sender
+        and Txn.amount == 0
+        and Txn.rekey_to == Global.zero_address
+        and Txn.close_remainder_to == Global.zero_address
+        and Txn.fee == 2 * Global.min_txn_fee
+        and Txn.last_valid <= TemplateVar[UInt64]("EXPIRATION_ROUND")
```

Comment:
Isn't a `lease` needed here?

Reply:
The goal of the example is how you can provide fees for all calls to an app as a Contract Account.

I don't think so? Every call to this dummy app is legit. And if the same app call is replayed, then it will be rejected by the same TxID, right?

Reply:
Yes exactly.